### ISSUE TITLE
Brain IS IDIOT

### ISSUE THREAD

**Issue by ghost** [BrainJS/brain.js#682]
## _What_ is wrong?
BRAIN.JS really have autism or something

## _Where_ does it happen?
I train him to give me numbers, he output me > 1 float every time

## _How_ do we replicate the issue?

1 )
`var trainingData = 
{
	input:[0],output:[3],
	input:[1],output:[3],
	input:[2],output:[3],
	input:[3],output:[3],
	input:[4],output:[3],
	input:[5],output:[1],
	input:[6],output:[3],
	input:[7],output:[2],
	input:[8],output:[1],
	input:[9],output:[1],
}`
2)  Train & Load in new js script
3) 
`const net = new brain.NeuralNetworkGPU();
let fileContent = fs.readFileSync("model.json",'utf8');
let fileContentJSON= JSON.parse(fileContent);
net.fromJSON(fileContentJSON);
console.log(net.run([8]));`
Supposed output : 1

## _How_ important is this ?   4

![image](https://user-images.githubusercontent.com/52961795/104140023-7c9d5500-53af-11eb-80e4-fae4126ab5c3.png)

**Comment by robertleeplummerjr** [comment]
It looks like from your training data you are giving some very weird training data.  It should be an array of objects in this case.

**Comment by ghost** [comment]
Could you give m an example of Array of Objects using my model.
Im trying to learn my IA to play "One Two Ten" game.
Each player each round chose une number between 1 and 3 and the main number get added, the objective is to force the other player to say the number 10.
Points : 0
Player1 : 3 (3)
Player2 : 3 (6)
Player1 : 1 (7)
Player2 : 2 (9)
Player1 : 1 (10)

Player 2 won.

**Comment by robertleeplummerjr** [comment]
The working example: https://observablehq.com/@robertleeplummerjr/brain-js-issue-682
In Neural Network terms, in dealing with floating point and network accuracy, `0.98` IS `1`.  The exception is when using a Recurrent Neural Network, such as `brain.recurrent.LSTM`, where inputs are associated with a neuron.  It sounds like you want Recurrent Neural Network for the task at hand.  Also note, that your inputs and outputs seem to overwrite each other, but it could b a typo. Generally in training you supply an array of training values.

In a Recurrent Neural Network (note: *NOT* Time Step Recurrent Neural Network) `"5"` (string) or `"9"` (string) is a string, associated with a neuron, which is combined with all other characters that the neural network accepts as a matrix such as this:

In the example of `"5"` (string), and the network "speaks" 0-9, then the initial input actually looks something like this:
```
[
  //0
  0,
  //1
  0,
  //2
  0,
  //3
  0,
  //4
  0,
  //5
  1, // note: 5 is activated
  //6
  0,
  //7
  0,
  //8
  0,
  //9
  0,
  //unknown
  0,
]
```

In the example of `"9"` (string), and the network "speaks" 0-9, then the initial input actually looks something like this:
```
[
  //0
  0,
  //1
  0,
  //2
  0,
  //3
  0,
  //4
  0,
  //5
  0,
  //6
  0,
  //7
  0,
  //8
  0,
  //9
  1, // note: 9 is activated
  //unknown
  0,
]
```

**Comment by ghost** [comment]
I am just trying to understand now, i just replaced 
const net = new brain.NeuralNetworkGPU();
by
const net = new brain.recurrent.LSTM();

and now its working fine. What is the difference.

**Comment by robertleeplummerjr** [comment]
>What is the difference.

NeuralNetwork is a feed forward neural network.  How it works: Imagine selecting a single frame of a movie.  What can you tell me about the movie frame?  Likely only the things in it, maybe the colors, where it takes place, etc.

LSTM is a Long Short Term Memory Recurrent Neural Network.  How it works: Imagine each frame of a movie played together over time.  What can you tell me about the movie?  Not only can we identify things, colors, places, but now we have context _over time_.  We can derive that there are dialog, jokes, drama, sadness, happiness, old characters, new characters, etc.
